notebook/agentchat_lmm_llava.ipynb

```diff
@@ -0,0 +1,1118 @@
+   "outputs": [],
+   "source": [
+    "import replicate"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1805e4bd",
+   "metadata": {},
+   "source": [
+    "## [Option 2] Setup LLaVA Locally\n",
+    "Please follow the LLaVA GitHub [page](https://github.com/haotian-liu/LLaVA/) to install LLaVA, download the weights, and start the server.\n",
+    "\n",
+    "For instance, here are some important steps:\n",
+    "```bash\n",
+    "# Download the package\n",
+    "git clone https://github.com/haotian-liu/LLaVA.git\n",
+    "cd LLaVA\n",
+    "\n",
+    "# Install the inference package\n",
+    "conda create -n llava python=3.10 -y\n",
+    "conda activate llava\n",
+    "pip install --upgrade pip  # enable PEP 660 support\n",
+    "pip install -e .\n",
+    "\n",
+    "# Download and serve the model\n",
+    "python -m llava.serve.model_worker --host 0.0.0.0 --controller http://localhost:10000 --port 40000 --worker http://localhost:40000 --model-path liuhaotian/llava-v1.5-7b\n",
+    "```\n",
+    "\n",
+    "Some helpful packages and dependencies:\n",
+    "```bash\n",
+    "conda install -c nvidia cuda-toolkit\n",
+    "```\n",
+    "\n",
+    "\n",
+    "### Launch\n",
+    "\n",
+    "In one terminal, start the controller first:\n",
+    "```bash\n",
+    "python -m llava.serve.controller --host 0.0.0.0 --port 10000\n",
+    "```\n",
+    "\n",
+    "\n",
+    "Then, in another terminal, start the worker, which will load the model to the GPU:\n",
+    "```bash\n",
+    "python -m llava.serve.model_worker --host 0.0.0.0 --controller http://localhost:10000 --port 40000 --worker http://localhost:40000 --model-path liuhaotian/llava-v1.5-13b\n",
+    "``"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9c29925f",
+   "metadata": {},
+   "source": [
+    "**Note: make sure the environment of this notebook also installed the llava package from `pip install -e .`**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "93bf7915",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[2023-10-18 13:49:01,342] [INFO] [real_accelerator.py:110:get_accelerator] Setting ds_accelerator to cuda (auto detect)\n",
+      "{'models': ['llava-v1.5-13b']}\n",
+      "Model Name: llava-v1.5-13b\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Run this code block only if you want to run LlaVA locally\n",
+    "import requests\n",
+    "import json\n",
+    "from llava.conversation import default_conversation as conv\n",
+    "from llava.conversation import Conversation\n",
+    "\n",
+    "# Setup some global constants for convenience\n",
+    "# Note: make sure the addresses below are consistent with your setup in LLaVA \n",
+    "CONTROLLER_ADDR = \"http://0.0.0.0:10000\"\n",
+    "SEP =  conv.sep\n",
+    "ret = requests.post(CONTROLLER_ADDR + \"/list_models\")\n",
+    "print(ret.json())\n",
+    "MODEL_NAME = ret.json()[\"models\"][0]\n",
```

Comment:
MODEL_NAME is also needed for "remote" code path

Reply:
Good catch! I will set `MODEL_NAME` to empty string at the beginning.